zerver/lib/streams.py

```diff
@@ -999,6 +999,17 @@ def add_filter_option(option: Q) -> None:
     return list(streams)
 
 
+def get_group_setting_value_dict_for_streams(
+    streams: list[Stream],
+) -> dict[int, int | AnonymousSettingGroupDict]:
+    setting_group_ids = set()
+    for stream in streams:
+        for setting_name in Stream.stream_permission_group_settings:
+            setting_group_ids.add(getattr(stream, setting_name + "_id"))
+
+    return get_setting_values_for_group_settings(list(setting_group_ids))
+
+
 def get_setting_values_for_group_settings(
```

Comment:
Have still kept this function because at a couple of places where we have stream dicts and not Stream objects.